File: Renta.Apps.Common/Models/PagedList.cs
using System.Collections.Generic;
using System.Linq;
using Renta.Apps.Common.Interfaces;

namespace Renta.Apps.Common.Models
{
    public class PagedList<T> : IPagedList<T>
    {
        private T[] _items;
        
        public PagedList()
        {
        }

        public PagedList(int pageNumber, int pageSize)
        {
            _items = new T[0];
            PageNumber = pageNumber;
            PageSize = pageSize;
        }

        public PagedList(IEnumerable<T> items, int pageNumber, int pageSize, int totalItemCount)
        {
            _items = items?.ToArray() ?? new T[0];
            PageNumber = pageNumber;
            PageSize = pageSize;
            TotalItemCount = totalItemCount;
        }
        
        public int TotalItemCount { get; set; }
        
        public int PageNumber { get; set; }
        
        public int PageSize { get; set; }

        public int PageCount => ((TotalItemCount > 0) && (PageSize > 0)) ? ((TotalItemCount - 1) / PageSize + 1) : 1;

        public T[] Items
        {
            get => _items ??= new T[0];
            set => _items = value;
        }
    }
}

File: ReApptor.TypeScript.PagedList/src/models/PagedList.ts
import IPagedList from "../interfaces/IPagedList";
import PagedListUtility from "../utilities/PagedListUtility";

export default class PagedList<T = {}> implements IPagedList<T> {
    
    public items: T[] = [];
    
    public totalItemCount: number = 0;

    public pageNumber: number = 0;

    public pageSize: number = 0;

    public get pageCount(): number {
        return ((this.totalItemCount > 0) && (this.pageSize > 0)) ? ((this.totalItemCount - 1) / this.pageSize + 1) : 1;
    }
    
    public isPagedList: true = true;

    public static toPagedList<T>(items: T[], pageNumber: number, pageSize: number): IPagedList<T> {
        return PagedListUtility.toPagedList<T>(items, pageNumber, pageSize);
    }
}